src/reason_parser.mly

```diff
@@ -2491,6 +2522,136 @@ labeled_simple_pattern:
   | simple_pattern
       { ("", None, $1) }
 ;
+jsx_arguments:
+  /* empty */ { [] }
+  | LIDENT EQUAL simple_expr jsx_arguments
+      {
+        (* a=b *)
+        [($1, $3)] @ $4
+      }
+  | LIDENT jsx_arguments
+      {
+        (* a (punning) *)
+        let loc_lident = mklocation $startpos($1) $endpos($1) in
+        [($1, mkexp (Pexp_ident {txt = Lident $1; loc = loc_lident}))] @ $2
+      }
+;
+
+jsx_name:
+  | mod_longident { $1 }
+  | LIDENT { Lident $1 }
+;
+
+
+jsx_start_tag_body:
+  | jsx_name jsx_arguments {
+      (jsx_component $1 $2, $1)
+  }
+;
+
+jsx_separate_open_close:
+  | jsx_start_tag_body GREATERLESSSLASH jsx_name
+      {
+        (* Foo></Foo *)
+        let (component, start) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        let _ = ensureTagsAreEqual start $3 loc in
+        component [("", mktailexp_extension loc [] None)] ~loc
+      }
+  | jsx_start_tag_body GREATERLESS jsx_tag_siblings jsx_name
```

Comment:
This seems to allow for there to be a space between `<` and the tag name. `< TagName`. I wonder if things become simpler if the lexer requires that there be no space.